File: AUGUST-2025/05-08-2025/p3477.c
int numOfUnplacedFruits(int* fruits, int fruitsSize, int* baskets, int basketsSize) {
    int count = 0;
    for(int i = 0; i < fruitsSize; i++)
    {
        int unset = 1;
        for(int j = 0; j < basketsSize; j++)
        {
            if(fruits[i] <= baskets[j])
            {
                baskets[j] = 0;
                unset = 0;
                break;
            }
        }
        count += unset;
    }
    return count;
}

File: AUGUST-2025/05-08-2025/p3477.scala
object Solution {
    def numOfUnplacedFruits(fruits: Array[Int], baskets: Array[Int]): Int = {
        fruits.foldLeft(baskets.zipWithIndex) {
            case (acc, value) =>
                acc.find {
                    case (v, _) => v >= value
                } match {
                    case None => acc
                    case Some((v, idx)) =>
                        acc(idx) = (0, idx)
                        acc
                }
        }.count {
            case (v, _) => v > 0
        }
    }
}